tests/simulation/raft.cpp

```diff
@@ -229,6 +225,10 @@ void RunSimulation() {
 
   simulator.ShutDown();
 
+  srv_thread_1.join();
+  srv_thread_2.join();
+  srv_thread_3.join();
```

Comment:
`std::jthread`s are RAII objects, they join when their destructor is called.

Reply:
we want to join them before retrieving the simulator stats because we want to make sure that they have finished interacting with the simulator before getting the final message counts etc...